tests/sss/test_subscription_config.py

```diff
@@ -0,0 +1,32 @@
+import pytest
+
+from tests.testutils.addon_helpers import addon_with_imageset  # noqa: F401
+from tests.testutils.addon_helpers import addon_with_indeximage  # noqa: F401
+
+
+@pytest.mark.parametrize(
+    "addon_str",
+    [
+        "addon_with_indeximage",
+        "addon_with_imageset",
+    ],
+)
+def test_subscription_config(addon_str, request):
+    expected_values = [
+        {"name": "LOCATION", "value": "Black Mesa Research Facility"},
+        {"name": "USER", "value": "Gordon Freeman"},
+    ]
+    addon = request.getfixturevalue(addon_str)
+    sss_walker = addon.sss.walker()
+    subscription_obj = sss_walker["sss_deploy"]["spec"]["resources"][
+        "Subscription"
+    ][0]
+    if addon.subscription_config_present():
```

Comment:
Do we actually need this method? I see that its only being used in the test. We can also simply replace it with:
```suggestion
    if addon.metadata.get("config"):
```

Reply:
I think that method can stay, `subscription_config_present` abstracts the implementation detail. I actually forgot to use that in the jinja template, fixed it now.